docs/api/DesktopAgent.md

```diff
@@ -179,7 +179,7 @@ catch (err){
 ```ts
 getSystemChannels() : Promise<Array<Channel>>;
 ```
-Retrieves a list of the System channels available for the app to join.
+Retrieves a list of the System channels available for the app to join.  This should include the 'global' channel.
```

Comment:
@vmehtafds The biggest difference between 'global' and 'default' is that you can leave 'global'.

Reply:
@vmehtafds another reason for this is I think that the name `default` was part of the problem - the default behaviour is now not to be on a channel (unless you call `joinChannel`). The name `global` makes it clearer what this channel does, I think.

Reply:
> the default behaviour is now not to be on a channel

@rikoe: this is a long thread so i may have missed this somewhere, but how does this change not break implementors who were only utilizing `DesktopAgent.addContextListener(...)` / `DesktopAgent.broadcast(...)`?

I would question why any change is needed at all. If you never add a context listener, you are de-facto not in any channel, global or otherwise.

The new proposal seems to indicate that using the Channel join APIs are mandatory in order to utilize context; or am I missing something here?

Reply:
@nicholasgoodman @rikoe  Implementors wishing to maintain backwards compatibility with existing behavior with `desktopAgent.addContextListener` and `desktopAgent.broadcast` can choose to start apps on the `global` channel by default.  The key differences from previous proposals are:

- users can leave the `global` channel
- there is no special behavior to implement for the `global` channel
- naming is a lot clearer

Reply:
I would think that maintaining compatibility with the 1.0 spec would imply that auto-joining the global channel would be a SHOULD instead of a MAY; but either way it seems acceptable to me.

Additionally, not sure what "leaving all channels" provides for that can't be handled within the existing add listener or listener callback code; but similarly, I'm not against it either.

Reply:
@nicholasdgoodman   This is backwards compatible with the 1.0 *spec*.  However, 1.0 was very open ended on this topic  and many implementations currently work this way.  If you're building an implementation today - especially if your platform leans heavily on color (or other) linking - you may not want apps connected by default.  So, I'd rather keep this as a MAY.  

The "leaving all channels" pattern is a convenience - but it can be a pretty big one.  Apps may have multiple handlers set for context and they may be also broadcasting from multiple points.  Also, if some controller for linking is operating on the app's behalf (for example, a preload with a titlebar & color selector) it can turn on/off linking without needing knowledge of how listeners and broadcasters are wired up.